contracts/lease/src/position/spec.rs

```diff
@@ -87,6 +109,38 @@ where
         }
     }
 
+    /// Calculate the borrow amount.
+    /// Return 'error::ContractError::InsufficientTrasactionAmount' when either the downpayment
+    /// or the borrow amount is with amount less than the minimum trasaction amount.
+    /// Return 'error::ContractError::InsufficientAssetAmount' when the lease (downpayment + borrow)
+    /// is with amount less than the minimum asset amount.
+    pub fn calc_borrow_amount(
+        &self,
+        downpayment: Coin<Lpn>,
+        may_max_ltd: Option<Percent>,
+    ) -> ContractResult<Coin<Lpn>> {
+        let price: Price<Lpn, Lpn> = price::total_of(1.into()).is(1.into());
+
+        let _ = self
+            .check_trasaction_amount(downpayment, price)
+            .map_err(|err| match err {
+                ContractError::PositionCloseAmountTooSmall(min_amount) => {
```

Comment:
it is not OK `check_transaction_amount` to return an error related to PolitionClose, is it?